blant.c

```diff
@@ -1410,17 +1410,18 @@ int RunBlantFromGraph(int k, int numSamples, GRAPH *G)
     case graphletFrequency:
 	for(canon=0; canon<_numCanon; canon++) {
 	if (_sampleMethod == SAMPLE_MCMC && !_windowRep) {
-		BuildGraph(g, _canonList[canon]);
-		if (TinyGraphDFSConnected(g, 0)) {
+		if (SetIn(_connectedCanonicals, canon)) {
 			printf("%lf ", _graphletConcentration[canon]);
 			PrintCanonical(canon);
 			printf("\n");
 		}
 	}
 	else {
-		printf("%lu ", _graphletCount[canon]);
-		PrintCanonical(canon);
-		printf("\n");
+		if (SetIn(_connectedCanonicals, canon)) {
```

Comment:
Problem here is we have a macro somewhere called ALLOW_DISCONNECTED_GRAPHLETS. Perhaps that should be made into a global variable and checked here at run time?

Reply:
```suggestion
		if (ALLOW_DISCONNECTED_GRAPHLETS || SetIn(_connectedCanonicals, canon)) {
```
Changing it to this makes sense to me since all non MCMC sampling algorithms work with this #define.

Reply:
Should ALLOW_DISCONNECTED_GRAPHLETS be a command line argument rather than a #define?